packages/expo-cli/src/commands/permissions.ts

```diff
@@ -0,0 +1,416 @@
+        // TODO(Bacon): Add descriptions of what each permission is used for
+      };
+      return descriptions[Object.keys(Manifest.UnimodulePermissions)[this.index]];
+    },
+    initial: permissions,
+    hint:
+      '(Use <space> to select, <return> to submit, <a> to toggle, <i> to inverse the selection)',
+    message: `Select Android permissions`,
+    choices,
+  });
+
+  console.log('');
+
+  let answer: string[];
+
+  try {
+    answer = await prompt.run();
+  } catch (error) {
+    console.log(chalk.yellow(`${CHEVRON} Exiting...`));
+    return;
+  }
+
+  const selectedAll = choices.length === answer.length;
+
+  console.log(
+    chalk.magenta(
+      `${CHEVRON} Saving selection to the ${chalk.underline`expo.android.permissions`} array in the ${chalk.bold`app.json`}...`
+    )
+  );
+  if (isExpo) {
+    if (selectedAll) {
+      console.log(
+        chalk.magenta(
+          `${CHEVRON} Expo will default to using all permissions in your project by deleting the ${chalk.underline`expo.android.permissions`} array.`
+        )
+      );
+    }
+  }
+
+  await writeConfigJsonAsync(projectDir, {
+    android: {
+      ...(exp.android || {}),
+      // An empty array means no permissions
+      // No value means all permissions
+      permissions: selectedAll
+        ? undefined
+        : answer.map((permission: string) => {
+            if (permission.startsWith('android.permission.')) {
+              return permission.split('.').pop();
+            }
+            return permission;
+          }),
+    },
+  });
+
+  if (!isExpo) {
+    console.log(
+      chalk.magenta(`${CHEVRON} Saving selection to the native ${chalk.bold`AndroidManifest.xml`}`)
+    );
+
+    await Manifest.persistAndroidPermissionsAsync(projectDir, [
+      'android.permission.INTERNET',
+      ...answer,
+    ]);
+  }
+}
+
+type IOSPermissionChoice = { name: string; message: string; initial: string };
+
+async function promptForPermissionDescriptionsAsync(
+  choices: IOSPermissionChoice[],
+  hasNativeConfig: boolean,
+  currentDescriptions: AnyPermissions
+): Promise<AnyPermissions> {
+  console.log('');
+
+  const keys = Object.keys(DefaultiOSPermissionNames);
+  const prompt = new Form({
+    name: hasNativeConfig ? 'Native project' : 'Universal Project',
+    message: 'Modify iOS Permissions',
+    header() {
+      const permission = keys[this.index];
+      if (!currentDescriptions[permission]) {
+        return chalk.magenta(
+          `${CHEVRON} Add a description to enable the "${chalk.bold(
+            permission
+          )}" permission in your iOS app`
+        );
+      }
+      return chalk.magenta(
+        `${CHEVRON} Update the description for the permission "${chalk.bold(permission)}"`
+      );
+    },
+    choices,
+  });
+
+  try {
+    const answer: AnyPermissions = await prompt.run();
+    // Trimming allows users to pass in " " to delete a permission
+    return Object.keys(answer).reduce(
```

Comment:
simpler:
```
for (const key of Object.keys(answer)) {
  answer[key] = answer[key].trim();
}
```

or https://lodash.com/docs/2.4.2#mapValues. mapValues seems particularly useful to remove all the reduce() calls here